lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_perform_step.jl

```diff
@@ -1691,159 +1691,42 @@ end
-    else
-        @.. broadcast=false du1=dtC[5][2] * ks[2] + dtC[5][4] * ks[4] + dtC[5][1] * ks[1] + dtC[5][3] * ks[3]
-        mul!(_vec(dus[3]), mass_matrix, _vec(dus[2]))
-        @.. broadcast=false linsolve_tmp=dus[1] + dtd[5] * dT + dus[3]
-    end
-
-    linres = dolinsolve(integrator, linres.cache; b = _vec(linsolve_tmp))
-    veck5 = _vec(ks[5])
-    @.. broadcast=false veck5=-vecu
-    integrator.stats.nsolve += 1
-
-    @.. broadcast=false u=uprev + a[6][1] * ks[1] + a[6][2] * ks[2] + a[6][3] * ks[3] + a[6][4] * ks[4] + a[6][5] * ks[5]
-    stage_limiter!(u, integrator, p, t + dt)
-    f(du, u, p, t + dt)
-    integrator.stats.nf += 1
-
-    if mass_matrix === I
-        @.. broadcast=false linsolve_tmp=du + (dtC[6][1] * ks[1] + dtC[6][2] * ks[2] + dtC[6][3] * ks[3] +
-                                          dtC[6][4] * ks[4] + dtC[6][5] * ks[5])
-    else
-        @.. broadcast=false du1=dtC[6][1] * ks[1] + dtC[6][2] * ks[2] + dtC[6][3] * ks[3] + dtC[6][4] * ks[4] +
-                                dtC[6][5] * ks[5]
-        mul!(_vec(dus[3]), mass_matrix, _vec(dus[2]))
-        @.. broadcast=false linsolve_tmp=du + dus[3]
-    end
-
-    linres = dolinsolve(integrator, linres.cache; b = _vec(linsolve_tmp))
-    veck6 = _vec(ks[6])
-    @.. broadcast=false veck6=-vecu
-    integrator.stats.nsolve += 1
+    for i in 1:8
+        if i == 1 && repeat_step
+            linres = dolinsolve(
+                integrator, cache.linsolve; A = nothing, b = _vec(linsolve_tmp),
+                du = cache.fsalfirst, u = u, p = p, t = t, weight = weight,
+                solverdata = (; gamma = dtgamma))
+        else
+            linres = dolinsolve(integrator, cache.linsolve; A = W, b = _vec(linsolve_tmp),
+                du = cache.fsalfirst, u = u, p = p, t = t, weight = weight,
+                solverdata = (; gamma = dtgamma))
+        end
 
-    u .+= ks[6]
-    stage_limiter!(u, integrator, p, t + dt)
-    f(du, u, p, t + dt)
-    integrator.stats.nf += 1
+        vecu = _vec(linres.u)
+        veck = _vec(ks[i])
 
-    if mass_matrix === I
-        @.. broadcast=false linsolve_tmp=dus[1] + (dtC[7][1] * ks[1] + dtC[7][2] * ks[2] + dtC[7][3] * ks[3] +
-                                          dtC[7][4] * ks[4] + dtC[7][5] * ks[5] + dtC[7][6] * ks[6])
-    else
-        @.. broadcast=false du1=dtC[7][1] * ks[1] + dtC[7][2] * ks[2] + dtC[7][3] * ks[3] + dtC[7][4] * ks[4] +
-                                dtC[7][5] * ks[5] + dtC[7][6] * ks[6]
-        mul!(_vec(dus[3]), mass_matrix, _vec(dus[2]))
-        @.. broadcast=false linsolve_tmp=du + dus[3]
-    end
+        @.. broadcast=false veck=-vecu
+        integrator.stats.nsolve += 1
 
-    linres = dolinsolve(integrator, linres.cache; b = _vec(linsolve_tmp))
-    veck7 = _vec(ks[7])
-    @.. broadcast=false veck7=-vecu
-    integrator.stats.nsolve += 1
+        @.. broadcast=false u=uprev + sum(a[i, j] * ks[j] for j in 1:i-1)
```

Comment:
This will remove the allocation
```suggestion
        @..  u=uprev
        for j in 1:i-1
            @.. u  += a[i, j] * ks[j]
       end
```